test/Generator.Tests/MapDateOnlyConverter.UnitTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Generator.Tests;
+
+[TestClass]
+public class MapDateOnlyConverterUnitTests
+{
+    private readonly JsonSerializerOptions options = new JsonSerializerOptions { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull };
+
+    [TestMethod]
+    public void PopulatedMapDateOnlyPropertyDeserializesCorrectly()
+    {
+        var json = $"{{\"$dtId\":\"d8985302-4ee1-4a10-b2f5-e854e1682422\",\"assetTag\":\"12345\",\"name\":\"Test Asset\",\"serialNumber\":\"SN12345\",\"runtimeDetails\":{{\"Friday\":\"2020-03-10\",\"Sunday\":\"2020-03-12\"}},\"$metadata\":{{\"$model\":\"{Asset.ModelId}\"}}}}";
+        var expectedAsset = new Asset
+        {
+            Id = "d8985302-4ee1-4a10-b2f5-e854e1682422",
+            AssetTag = "ABC123",
+            Name = "Test Asset",
+            SerialNumber = "SN12345",
+            RuntimeDetails = new Dictionary<string, DateOnly>()
+            {
+                ["Friday"] = new DateOnly(2020, 03, 10),
+                ["Sunday"] = new DateOnly(2020, 03, 12)
+            }
+        };
+        var deserializedAsset = JsonSerializer.Deserialize<Asset>(json, options);
+        Assert.AreEqual(expectedAsset.Id, deserializedAsset?.Id);
+        Assert.AreEqual(expectedAsset.Name, deserializedAsset?.Name);
+        Assert.AreEqual(expectedAsset.SerialNumber, deserializedAsset?.SerialNumber);
+        Assert.AreEqual(expectedAsset.RuntimeDetails["Friday"], deserializedAsset?.RuntimeDetails!["Friday"]);
```

Comment:
Rule of thumb for string literals...
1. If you're using the same string literal in multiple files, make it a project/solution-level constant.
2. If you're using the same string literal multiple times in the same file, make it a file-level constant.